### DISCUSSION TITLE
how to response bytes instead of json?

### DISCUSSION THREAD

**Discussion by crazy-canux** [fastapi/fastapi#7494] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [ ] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [ ] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [ ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [ ] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

when I set "content-type" in response header as something else instead of "application/json".
why the response header always have "content-type=application/json"?

```Python
from fastapi import FastAPI

app = FastAPI()

@app.get("/v2/{name:path}/manifests/{reference}", status_code=200)
async def get_manifests(response: Response, name, reference, request: Request):
    logger.debug(request.headers)
    file_name = os.path.join("/my/registry", "v2",  name, "manifests", reference)
    with open(file_name, "rb") as f:
        chunk = f.read()
    logger.debug(len(chunk))
    sha256 = hashlib.sha256(chunk).hexdigest()
    response.headers["Docker-Distribution-Api-Version"] = "registry/2.0"
    response.headers["Docker-Content-Digest"] = "sha256:{}".format(sha256)
    response.headers["Content-Type"] = "application/vnd.docker.distribution.manifest.v2+json"
    #response.headers["Content-Length"] = str(len(chunk))
    response.status_code = status.HTTP_200_OK
    logger.debug(response.headers)
    return chunk
```

### Description

If I set the "content-length" with len(chunk).
Got this errors:
`net/http: HTTP/1.x transport connection broken: http: message cannot contain multiple Content-Length headers; got ["1236" "1571"]`
I think it's because the response content-type transfer to "application/json" by fastapi, makes the content-length changed.

If not set "content-length", let's fastapi calculate by default.
The registry will not working. also because of the "content-type" is not "application/vnd.docker.distribution.manifest.v2+json".

![image](https://user-images.githubusercontent.com/14313415/85219284-087c6f00-b3d5-11ea-9c0e-2c327ad0f3bf.png)

### Environment

* OS: [e.g. Linux / Windows / macOS]:    docker alpine
* FastAPI Version [e.g. 0.3.0]:                   0.58.0

To know the FastAPI version use:

```bash
python -c "import fastapi; print(fastapi.__version__)"
```

* Python version:   python3.6

To know the Python version use:

```bash
python --version
```

### Additional context

How to return bytes with specify "content-type"?
How to disable the default  "content-type: application/json" ?

**Comment by crazy-canux** [comment, marked as answer]
fixed by using FileResponse.

**Comment by tiangolo** [comment]
Thanks for reporting back and closing the issue :+1:
